core/runtime/impl/tagged_transaction_queue_impl.cpp

```diff
@@ -10,13 +10,20 @@ namespace kagome::runtime {
 
   TaggedTransactionQueueImpl::TaggedTransactionQueueImpl(
       common::Buffer state_code,
-      std::shared_ptr<extensions::Extension> extension)
-      : RuntimeApi(std::move(state_code), std::move(extension)) {}
+      std::shared_ptr<extensions::Extension> extension,
+      std::shared_ptr<blockchain::BlockTree> block_tree,
+      std::shared_ptr<blockchain::BlockHeaderRepository> header_backend)
+      : RuntimeApi(std::move(state_code), std::move(extension)),
+        block_tree_(std::move(block_tree)),
+        header_backend_(std::move(header_backend)) {}
```

Comment:
please add boost asserts that shared pointers are not null

Reply:
It makes no sense, since I'm going to remove this dependencies.

Reply:
It makes no sense here, since no additional sptrs will be declared in constructor. Extension sptr should be handled by RuntimeApi. However, I added boost asserts in extrinsic api impl constructor.